Added documentation for the "native" DF-MP2 implementation. (#79)
pyscf/pyscf.github.io: source/user/mp.rst

Introduction
============

> Second-order Møller–Plesset perturbation theory (MP2) :cite:`Moller1934` is a

Reviewer: ```suggestion
Second-order Møller--Plesset perturbation theory (MP2) :cite:`Moller1934` is a
```

> post-Hartree--Fock method.  MP2 calculations can be performed with or without
> density fitting, depending on the initial SCF calculation. Please make sure to
> read the section on the :ref:`native DF-MP2 implementation<DFMP2>`, which does
> not depend on the integral approximation in SCF.

Reviewer: ```suggestion
post-Hartree--Fock method. MP2 calculations can be performed in PySCF with or without
density fitting, depending on the initial SCF calculation.

Note also the existence of a :ref:`native DF-MP2 implementation<DFMP2>`, which does
not depend on the integral approximation in SCF, and which is significantly faster than the
default implementation of MP2 with density fitting.
```

    mymp.kernel(with_t2=False)

.. _DFMP2:

Density-fitted MP2 (DF-MP2)
===========================

Background
----------

> MP2 can be combined to great benefit with the density fitting (DF) or resolution of the identity

Reviewer: ```suggestion
MP2 can be combined to great benefit with density fitting (DF), also known as the resolution of the identity
```

(RI) approximation. While the formal scaling remains :math:`O(N^5)`, the prefactor and the overall
computational cost are reduced strongly, so that calculations can be performed on much larger

> molecules than with conventional MP2. In general, the errors in reaction energies, geometries,
> properties etc. with a suitable auxiliary set are negligible compared to the intrinsic errors of
> MP2. This implementation can calculate energies, as well as unrelaxed and relaxed one-particle
> density matrices. Analytical gradients are not available yet. RHF and UHF references are supported
> throughout.

Reviewer: ```suggestion
molecules than with conventional MP2. Provided that a suitable auxiliary basis set is used,
the resulting DF/RI errors in reaction energies, geometries, properties etc. are negligible
compared to the intrinsic errors of MP2. 

This implementation can calculate energies, as well as unrelaxed and relaxed one-particle
density matrices. Analytical gradients are not available yet. RHF and UHF references are supported
throughout.
```

be imported from the respective modules ``mp.dfmp2_native`` and ``mp.dfump2_native``::

    from pyscf.mp.dfmp2_native import DFRMP2
    from pyscf.mp.dfump2_native import DFUMP2

Both modules also make the respective classes available under the alias ``DFMP2``. The file
:source:`examples/mp/10-dfmp2.py` contains a simple example for a DF-MP2 energy calculation.

There is also an older variant of DF-MP2 implemented in ``mp.dfmp2``, which does not exploit the
advantages of density fitting fully, and may therefore have much higher memory demands for larger
molecules. Please note that the ``density_fit()`` method of the conventional ``mp.mp2.MP2`` class
provides an instance of the old implementation, not the new one! Currently, the "native"
implementation can only be used by importing it from ``mp.dfmp2_native`` or ``mp.dfump2_native``
directly as shown above.

Unless specified by the user, an appropriate auxiliary basis set is determined automatically. Note
that there exist different auxiliary basis sets for Coulomb and exchange fitting in DF-HF on the one
hand, and for correlation fitting in MP2 or other dynamic correlation methods on the other hand.

> Anyway, the DF approximation in MP2 should not depend on the approximation taken for SCF. Arbitrary

Reviewer: ```suggestion
The DF approximation in MP2 does not depend on the approximation taken for SCF. Arbitrary
```

auxiliary sets can be specified with the ``auxbasis`` option::

    DFMP2(mf, auxbasis='cc-pVTZ-RI')

> In RHF-DF-MP2 calculations, orbitals can be frozen by specifying an integer or a list. Frozen core
> UHF-DF-MP2 calculations are initiated by providing an integer or two lists of equal length (for the
> alpha and beta orbitals)::

Reviewer: ```suggestion
In RHF-DF-MP2 calculations, orbitals can be frozen by specifying either an integer, or a list. Frozen core
UHF-DF-MP2 calculations are initiated by providing either an integer, or two lists of equal length (for the
alpha and beta orbitals)::
```
